crates/payload/basic/src/lib.rs

```diff
@@ -535,199 +538,215 @@ enum BuildOutcome {
     Cancelled,
 }
 
-/// Builds the payload and sends the result to the given channel.
-fn build_payload<Pool, Client>(
+struct BuildArguments<Pool, Client> {
```

Comment:
this and the fields also need to be public + docs

Reply:
@mattsse Done